share/eds/cache/accessor_cache.go

```diff
@@ -0,0 +1,194 @@
+package cache
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"reflect"
+	"sync"
+
+	"github.com/filecoin-project/dagstore"
+	"github.com/filecoin-project/dagstore/shard"
+	lru "github.com/hashicorp/golang-lru"
+)
+
+var _ Cache = (*AccessorCache)(nil)
+
+// AccessorCache implements the Cache interface using an LRU cache backend.
+type AccessorCache struct {
+	// The name is a prefix that will be used for cache metrics if they are enabled.
+	name string
+	// stripedLocks prevents simultaneous RW access to the blockstore cache for a shard. Instead
+	// of using only one lock or one lock per key, we stripe the shard keys across 256 locks. 256 is
+	// chosen because it 0-255 is the range of values we get looking at the last byte of the key.
+	stripedLocks [256]sync.Mutex
+	// Caches the blockstore for a given shard for shard read affinity, i.e., further reads will likely
+	// be from the same shard. Maps (shard key -> blockstore).
+	cache *lru.Cache
+
+	metrics *metrics
+}
+
+// accessorWithBlockstore is the value that we store in the blockstore Cache. It implements the
+// Accessor interface.
+type accessorWithBlockstore struct {
+	sync.RWMutex
+	shardAccessor Accessor
+	// The blockstore is stored separately because each access to the blockstore over the shard
+	// accessor reopens the underlying CAR.
+	bs dagstore.ReadBlockstore
+}
```

Comment:
[nit]
move methods closer to the struct

Reply:
This is core struct for this file, so I moved it to the top to ease code reading.

Reply:
I'm confused tbh, could you suggest the order you think would be better? Currently, methods declared in order of declared types:
1. AccessorCache type declaration
2. accessorWithBlockstore  type declaration
3. AccessorCache methods
4. accessorWithBlockstore  methods

Reply:
My nit suggestion is to swap 3 and 4, because accessorWithBlockstore is core and has only two small methods